tests/ops/api/v1/endpoints/test_dataset_endpoints.py

```diff
@@ -412,6 +416,413 @@ def test_put_validate_dataset(
+        response = api_client.patch(datasets_url, headers=auth_header, json=payload)
+
+        assert 422 == response.status_code
+        assert (
+            json.loads(response.text)["detail"][0]["msg"]
+            == "ensure this value has at most 50 items"
+        )
+
+    def test_patch_create_datasets_bulk_create(
+        self,
+        ctl_dataset,
+        generate_auth_header,
+        api_client,
+        datasets_url,
+        db,
+        request_body,
+    ):
+        request_body.append(
+            {
+                "fides_key": "second_dataset_config",
+                "ctl_dataset_fides_key": ctl_dataset.fides_key,
+            }
+        )
+
+        auth_header = generate_auth_header(scopes=[DATASET_CREATE_OR_UPDATE])
+        response = api_client.patch(
+            datasets_url,
+            headers=auth_header,
+            json=request_body,
+        )
+
+        assert response.status_code == 200
+        response_body = json.loads(response.text)
+        assert len(response_body["succeeded"]) == 2
+        assert len(response_body["failed"]) == 0
+
+        first_dataset_config = DatasetConfig.get_by(
+            db=db, field="fides_key", value="test_fides_key"
+        )
+        assert first_dataset_config.ctl_dataset == ctl_dataset
+        assert (
+            response_body["succeeded"][0]["collections"]
+            == first_dataset_config.dataset["collections"]
+        )
+        assert response_body["succeeded"][0]["fides_key"] == ctl_dataset.fides_key
+        assert (
+            first_dataset_config.dataset["collections"]
+            == Dataset.from_orm(ctl_dataset).collections
+        )
+        assert len(first_dataset_config.dataset["collections"]) == 1
+
+        second_dataset_config = DatasetConfig.get_by(
+            db=db, field="fides_key", value="second_dataset_config"
+        )
+        assert (
+            response_body["succeeded"][1]["collections"]
+            == first_dataset_config.dataset["collections"]
+        )
+        assert response_body["succeeded"][1]["fides_key"] == ctl_dataset.fides_key
+        assert second_dataset_config.ctl_dataset == ctl_dataset
+        assert (
+            second_dataset_config.dataset["collections"]
+            == Dataset.from_orm(ctl_dataset).collections
+        )
+
+        first_dataset_config.delete(db)
+        second_dataset_config.delete(db)
+
+    def test_patch_update_dataset_configs(
+        self,
+        ctl_dataset,
+        generate_auth_header,
+        api_client,
+        datasets_url,
+        db,
+        request_body,
+        dataset_config,
+    ):
+
+        old_ctl_dataset_id = dataset_config.ctl_dataset.id
+        assert dataset_config.ctl_dataset == ctl_dataset
+        updated = dataset_config.updated_at
+
+        db.expunge(ctl_dataset)
+        make_transient(
+            ctl_dataset
+        )  # http://docs.sqlalchemy.org/en/rel_1_1/orm/session_api.html#sqlalchemy.orm.session.make_transient
```

Comment:
Interesting, I've never seen transient objects before. That's a cool way to reuse the object